src/elements/helpers/defaultTransform.ts

```diff
@@ -20,16 +21,27 @@ type FlexibleModelElement<FDesc extends FieldDescriptions<string>> = {
   assets?: string[];
 };
 
-export const transformElementDataIn = <
-  FDesc extends FieldDescriptions<string>
->(): TransformIn<FlexibleModelElement<FDesc>, FDesc> => ({ fields }) => {
-  return ({ ...fields } as unknown) as FieldNameToValueMap<FDesc>;
+export const transformElementDataIn = <FDesc extends FieldDescriptions<string>>(
+  transformRole: boolean
```

Comment:
Rather than create a new transformer, I modified the existing one to conditionally handle the role transform. This can probably be backported to a few other elements if we're happy with this approach.

Ideally, it'd be nice to write a series of transformers that we can composer, e.g.

```ts
createTransformIn(
  transformRole,
  transformIsMandatory(true),
  // ... additional transforms
)
```

to get code that's more reusable, but it's quite difficult to do typesafely, and is beyond the scope of this PR.

Reply:
I like the direction of travel here. Transformer code is fairly repetitive at the moment.